hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/FlushTableProcedure.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.procedure;
+
+import java.io.IOException;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.RegionReplicaUtil;
+import org.apache.hadoop.hbase.procedure.flush.MasterFlushTableProcedureManager;
+import org.apache.hadoop.hbase.procedure2.ProcedureStateSerializer;
+import org.apache.hadoop.hbase.procedure2.ProcedureSuspendedException;
+import org.apache.hadoop.hbase.procedure2.ProcedureYieldException;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hbase.thirdparty.com.google.protobuf.UnsafeByteOperations;
+
+import org.apache.hadoop.hbase.shaded.protobuf.ProtobufUtil;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.MasterProcedureProtos.FlushTableProcedureStateData;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.MasterProcedureProtos.FlushTableState;
+
+@InterfaceAudience.Private
+public class FlushTableProcedure extends AbstractStateMachineTableProcedure<FlushTableState> {
+  private static final Logger LOG = LoggerFactory.getLogger(FlushTableProcedure.class);
+
+  private TableName tableName;
+
+  private byte[] columnFamily;
+
+  public FlushTableProcedure() {
+    super();
+  }
+
+  public FlushTableProcedure(MasterProcedureEnv env, TableName tableName) {
+    this(env, tableName, null);
+  }
+
+  public FlushTableProcedure(MasterProcedureEnv env, TableName tableName, byte[] columnFamily) {
+    super(env);
+    this.tableName = tableName;
+    this.columnFamily = columnFamily;
+  }
+
+  @Override
+  protected LockState acquireLock(MasterProcedureEnv env) {
```

Comment:
Checked the code, this may not work as expected, as in the procedure scheduler, we will not schedule any other procedures for a given queue if a procedure has held the exclusive lock, even if your procedure does not require any locks.

So here, I suggest you still make it require a shared lock, and add a comment here to say that actually we do not need any lock here, and use another issue to add the support, as it requires changing the procedure scheduler, which is not related to the core feature here.

Reply:
Didn't noticed this before, thanks for reminding me. Good idea, I'll fix it.